src/Nethermind/Nethermind.Logging/PathUtils.cs

```diff
@@ -16,16 +16,51 @@
  * along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
  */
 
+using System;
+using System.Diagnostics;
 using System.IO;
 using System.Reflection;
 
 namespace Nethermind.Logging
 {
     public static class PathUtils
     {
-        public static string GetExecutingDirectory()
+        public static string ExecutingDirectory { get; }
+
+        static PathUtils()
         {
-            return Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);;
+            var process = Process.GetCurrentProcess();
+            if (process.ProcessName.StartsWith("dotnet", StringComparison.InvariantCultureIgnoreCase))
+            {
+                ExecutingDirectory = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
+                return;
+            }
+
+            ExecutingDirectory = Path.GetDirectoryName(process.MainModule.FileName);
+        }
+
+        public static string GetApplicationResourcePath(this string resourcePath, string overridePrefixPath = null)
+        {
+            if (string.IsNullOrWhiteSpace(resourcePath))
+            {
+                return string.IsNullOrWhiteSpace(overridePrefixPath) ? ExecutingDirectory :
```

Comment:
Why do we need this if? If resource path is empty than Path.Combine will just not combine? Or will it throw? It makes the code more complicated without a good reason?

If null makes it throw exception just replace with string.empty.